model_analyzer/result/result_manager.py

```diff
@@ -154,23 +137,44 @@ def create_tables(self, gpu_specific_metrics, non_gpu_specific_metrics):
             headers=inference_output_headers,
         )
 
-        gpu_output_headers = []
-        gpu_output_fields = self._config.gpu_output_fields
-        for gpu_output_field in gpu_output_fields:
-            if gpu_output_field in self.headers:
-                gpu_output_headers.append(self.headers[gpu_output_field])
-            elif gpu_output_field in self._gpu_metrics_to_headers:
-                gpu_output_headers.append(
-                    self._gpu_metrics_to_headers[gpu_output_field])
-            else:
-                raise TritonModelAnalyzerException(
-                    f'GPU output field "{gpu_output_field}", does not exist')
-        self._gpu_output_fields = gpu_output_fields
-
-        # Model GPU Metrics
-        self._add_result_table(table_key=self.model_gpu_table_key,
-                               title='Models (GPU Metrics)',
-                               headers=gpu_output_headers)
+        if not self._is_cpu_only:
```

Comment:
This function is starting to get bloated. Perhaps we should split it up into `_create_server_table`, `_create_inference_table`, and `_create_gpu_table`.